### PR TITLE
psu-img: Update repository firmware image directory

### PR DESCRIPTION
**Pull request by faisal-awada**
Removed firmware and MANIFEST files from device-images directory and added new directory format to add subdirectory with the PSU CCIN. The new format structure to have a CCIN directory and in the new directory shall have the MANIFEST and firmware image.

Added new 51DA and 51E9 directory along with FWR and updated MANIFEST files.

Change-Id: Ia5db24225f1809d94c8995103c28d405a1f5502a

### CODE REVIEW

File: 51DA/MANIFEST [hunk 1]
```diff
@@ -0,0 +1,3 @@
+purpose=xyz.openbmc_project.Software.Version.VersionPurpose.PSU
+version=393141313900
```
Comment (derekhoward55) [inline 51DA/MANIFEST]: how did you determine this version number?  looks like this corresponds to "19.1A.09", but I don't think the 00 is valid.  Was this at offset 0x10 in the image?

File: 51E9/MANIFEST [hunk 2]
```diff
@@ -0,0 +1,3 @@
+purpose=xyz.openbmc_project.Software.Version.VersionPurpose.PSU
+version=393141313900
```
Comment (derekhoward55) [inline 51E9/MANIFEST]: same question as above?

File: README.md [hunk 3]
```diff
@@ -1 +1,22 @@
-# device-images
\ No newline at end of file
+OpenBMC power supply firmware
+
+This directory contains P11 power supply firmware images supplied by the
```
Comment (derekhoward55) [inline README.md]: I would remove the "P11" text, so it doesn't need to be modified later in P12

File: README.md [hunk 4]
```diff
@@ -1 +1,22 @@
...
+vendors. Each PSU firmware should reside in a directory named as the PSU CCIN.
+The firmware image file prefix should be replaced with IBM PSU CCIN and postfix
```
Comment (derekhoward55) [inline README.md]: I would remove the "IBM" text, so other companies can use the same naming convention.

File: README.md [hunk 5]
```diff
@@ -1 +1,22 @@
...
+    the tarball into the PSU CCIN dir.
+OR 
+
+* - Add firmware file to PSU CCIN Dir and create file with the following
+    contents:
+    purpose=xyz.openbmc_project.Software.Version.VersionPurpose.PSU
+    version=003000390031
+    extended_version=model=2B1E,manufacturer=
+
+Note: Files don't exist in this directory will require addition in
```
Comment (derekhoward55) [inline README.md]: perhaps explain why

**Review by derekhoward55 — COMMENTED** [review]
See individual comments...

File: README.md (hunk 4 above)
Comment (smccarney) [inline README.md]: Just curious, why do they need to rename their image file if it is already in a sub-directory that has the CCIN?

File: README.md [hunk 6]
```diff
@@ -1 +1,22 @@
-# device-images
\ No newline at end of file
+OpenBMC power supply firmware
+
+This directory contains P11 power supply firmware images supplied by the
+vendors. Each PSU firmware should reside in a directory named as the PSU CCIN.
+The firmware image file prefix should be replaced with IBM PSU CCIN and postfix
+must remain as is. The PSU CCIN dir should contains the firmware file and
```
Comment (smccarney) [inline README.md]: Nit: "should contain"

File: README.md [hunk 7]
```diff
@@ -1 +1,22 @@
...
+MANIFEST file. The contents of CCIN dir can be generated in 2 ways:
+
+* - Use tool to Generate Tarball with PSU image and MANIFEST Script, the
```
Comment (smccarney) [inline README.md]: This is kind of a run-on sentence with some missing words.  Also, why is "Generate Tarball" capitalized?

Possible rewording:

Use the tool at phosphor-psu-code-mgmt/tools/generate-psu-tar to generated a tarball with the PSU image and MANIFEST file.  Then extract the tarball into the PSU CCIN directory.

File: README.md [hunk 8]
```diff
@@ -1 +1,22 @@
...
+    tool located at phosphor-psu-code-mgmt/tools/generate-psu-tar then extract
+    the tarball into the PSU CCIN dir.
+OR 
+
+* - Add firmware file to PSU CCIN Dir and create file with the following
```
Comment (smccarney) [inline README.md]: I assume they don't want the file to contain *exactly* those contents right?  I'd think the version and extended_version would vary?  Maybe say "create a MANIFEST file with contents similar to the following:" ?

File: README.md (hunk 5 above)
Comment (smccarney) [inline README.md]: I'm not sure what this sentence is saying.  Do you mean "Note: Adding new files to this repository will require changes to ..." ?

File: README.md (hunk 8 above)
Comment (faisal-awada) [reply README.md]: It should say: Add firmware file to PSU CCIN Dir and create MANIFEST file with  contents similar to the following:

Comment (faisal-awada) [reply README.md]: version and extended_version have same value if different phosphor-psu-code-manager will not proceed. I have asked the author about it (no reply).

File: README.md (hunk 4 above)
Comment (faisal-awada) [reply README.md]: I will remove IBM, should I replace CCIN with model number?

Comment (derekhoward55) [reply README.md]: Not sure that it matters, whatever it's called in inventory manager is fine

**Review by derekhoward55 — COMMENTED** [review]
is the level number 2D.2F.2E?  AEI stores the bytes in reverse order of what the device driver will return them apparently.  The level in the manifest file should be compared exactly to what is in fw_version in sysfs.

**Review by smccarney — APPROVED** [review]
The README.md looks fine to me.  I don't know much about the directory structure and manifest files, so I will defer to Derek's review on those.

**Comment by derekhoward55** [comment]
Ok looks like the manifest files' fw_version is correct:

Enter the file name: IBM_RANIER_BOOTLOAD_2300W_1A62C_2E.2F.2D.hex
Enter version offset: 54
String read:  b'2D2F2E'
convert to ascii_tring and reversed bytes:  2D2F2E
FW version:  453246324432

I'd add the exact image file name to the commit message so we know which AEI file is checked in, other than that looks good to me.

File: 51DA/MANIFEST (hunk 1 above)
Comment (faisal-awada) [reply 51DA/MANIFEST]: The FW level pulled at offset 0x36 for AEI, then converted to ascii and reversed bytes.

File: 51E9/MANIFEST (hunk 2 above)
Comment (faisal-awada) [reply 51E9/MANIFEST]: The latest FW IBM_RANIER_BOOTLOAD_2300W_1A62C_2E.2F.2D.hex will show version=453246324432

**Review by derekhoward55 — APPROVED** [review]
Looks good!

**Review by smccarney — APPROVED** [review]

**Pull request merged**